library/classes/X12837P.php

```diff
@@ -0,0 +1,1239 @@
+<?php
+/* X12837P Class
+ * 
+ * This program creates a 5010 837P file
+ *
+ * @package OpenEMR
+ * @author Rod Roark <rod@sunsetsystems.com>
+ * @author Stephen Waite <stephen.waite@cmsvt.com>
+ * @copyright Copyright (c) 2009 Rod Roark <rod@sunsetsystems.com>
+ * @copyright Copyright (c) 2018 Stephen Waite <stephen.waite@cmsvt.com>
+ * @link https://github.com/openemr/openemr/tree/master
+ * @license https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+require_once(dirname(__FILE__) . "/../invoice_summary.inc.php");
+
+use OpenEMR\Billing\Claim;
+
+class X12837P
+{
+    public function gen_x12_837($pid, $encounter, &$log, $encounter_claim = false)
```

Comment:
make this a static function (this means to not need to initiate a class container since this class holds no information):
public static function gen_x12_837($pid, $encounter, &$log, $encounter_claim = false)